Assets/Scripts/Enemy/EnemyFactory.cs

```diff
@@ -46,45 +50,71 @@ public static EnemyFactory Instance
 
     public EnemyFactory()
     {
-        // Define enemyInfo for each type.
-        _enemyInfo.Add( EnemyType.TURRET, new EnemyInfo(_defaultFunc, Globals.DirectTargeting, _defaultMove));
-        
-        // Set up the dictionary with methods.
-        _enemyPostSetups.Add(EnemyType.TURRET, CreateTurret);
+        AddTurretToRoster();
+        // AddHealerToRoster();
     }
-    
+
     public GameObject CreateEnemy(EnemyType enemyType, Vector3 position)
     {
         // Janky, find a better place for this. Unsure when the Player object is available.
         if (_defaultTarget == null) _defaultTarget = GameObject.FindGameObjectWithTag("Player");
 
         if (!Globals.enemyPrefabNames.ContainsKey(enemyType)) return null;
-        
+
         string enemyName = Globals.enemyPrefabNames[enemyType];
-        
+
         GameObject newEnemyObject = GameObject.Instantiate(Resources.Load(enemyName)) as GameObject;
         Transform enemyTransform = newEnemyObject.GetComponent<Transform>();
-        
+
         // TODO: change default vector to dynamically adjust height of enemy spawn so they don't spawn under the ground.
         enemyTransform.position = position + new Vector3(0, 1, 0);
-        
-        Weapon enemyWeapon = newEnemyObject.AddComponent<Weapon>();
-        enemyWeapon.Initialize(new BulletAttackBehaviour(EntityType.ENEMY), 1f);
 
         EnemyInfo enemyInfo = _enemyInfo[enemyType];
-        
+
+        Weapon enemyWeapon = newEnemyObject.AddComponent<Weapon>();
+        enemyWeapon.Initialize(enemyInfo.attackBehaviour, enemyInfo.fireRate);
+
         EnemyBehaviour eb = EnemyBehaviour.AddToGameObject(
-            newEnemyObject, 
-            _defaultTarget, 
+            newEnemyObject,
+            _defaultTarget,
             enemyInfo,
             enemyWeapon);
         if (_enemyPostSetups.ContainsKey(enemyType)) _enemyPostSetups[enemyType](eb);
 
         return newEnemyObject;
     }
-    
-    private void CreateTurret(EnemyBehaviour eb)
+
+    void AddHealerToRoster()
+    {
+        BulletAttackBehaviour bulletBehaviour = new BulletAttackBehaviour(EntityType.ENEMY);

```

Comment:
Can we not create multiple methods for this and keep this functionality in the constructor?

Reply:
For organizing sake, lets keep it in multiple methods